camayoc/tests/qpc/cli/test_credentials.py

```diff
@@ -161,6 +162,44 @@ def test_add_with_username_sshkeyfile(data_provider, qpc_server_config):
     )
 
 
+@pytest.mark.ssh_keyfile_path
+def test_add_with_username_sshkeyfile_fromfile(data_provider, qpc_server_config, tmp_path):
+    """Add an auth with username and sshkeyfile from file.
+
+    :id: 9fd6b032-f78e-4056-b32b-9bd90cb41131
+    :description: Add an auth entry providing the ``--name``, ``--username``
+        and ``--sshkeyfile`` options.
+    :steps: Run ``qpc cred add --name <name> --username <username>
+        --sshkeyfile <ssh_keyfile_path>``
+    :expectedresults: A new auth entry is created with the data provided as
+        input.
+    """
+    name = utils.uuid4()
+    username = utils.uuid4()
+    ssh_key_content = "Multi-Line\nOpenSSH Key\nFrom File\n"
+    ssh_file_path = tmp_path / "test_ssh_key"
+    with Path.open(ssh_file_path, "w") as ssh_file:
+        ssh_file.write(ssh_key_content)
```

Comment:
Since `ssh_file_path` is Path already, you can just:

```suggestion
    ssh_file_path.write_text(ssh_key_content)
```

We already use that pattern (instead of opening for writing) in two other places.